### ISSUE TITLE
Test Focus for Dropdown across browsers

### ISSUE THREAD

**Issue by LHinson** [patternfly/patternfly-next#1023]
We have gotten feedback the following feedback regarding dropdown: 
> No focus indicator for the toggle button so there is no way for keyboard navigation users to know the item has focus

In this issue, we want to test the default browser focus for simple dropdown across all browsers. This issue includes identifying any issues for the core implementations and creating issue to address those issues.

**Comment by jgiardino** [comment]
Just wanted to update this issue, that after reviewing a bit more in Edge, I found that the focus indication becomes visible if I the pseudo element that creates the border (i.e. the border is hiding the focus indication).

**Comment by mcoker** [comment]
At a glance, probably the least invasive way to fix this would be to assign a negative `z-index` to the pseudo element, to push the border behind the parent. This would only work currently with buttons that have a transparent background.

A more invasive way to fix it would be to use an actual `border` on the button itself. That would likely involve adjusting the button's `padding` to accommodate the new border width.

**Comment by jgiardino** [comment]
If the button needs a background, could that be defined on the pseudo element with the border?

**Comment by mcoker** [comment]
@jgiardino yep, that should work!

**Comment by mcoker** [comment]
@christiemolloy FYI - reated https://github.com/patternfly/patternfly-next/issues/663

**Comment by christiemolloy** [comment]
@mcoker @jgiardino after reading the comments on this issue, I just want to clarify what to do here. 
Has this issue turned into fixing the focus style in Edge for buttons? So putting a negative z-index on the pseudo element so the border is behind the focus style, and also defining the background to a pseudo element so that this works for buttons that don't have a transparent background? 
Should I be doing anything for dropdowns and inputs too?
And from above it sounds like there are issues to fix focus styles in Firefox?

**Comment by jgiardino** [comment]
For this issue, I suggest reviewing and addressing issues we have in both Firefox and Edge for any components that use buttons. 

In #663, we intentionally added the styles that removed the default focus indicator in Firefox for all buttons. I'm not sure what the original issue was. For example, were the buttons in Firefox different in size compared to other buttons in FF or compared to buttons in other browsers? It might be worth chatting with @mcoker about that. If we have to choose between difference in size across browsers versus not indicating focus, I'd opt for the former. But @mcoker might have an alternate suggestion for this.

In Edge, the issue was most noticeable for secondary and tertiary buttons. In the dropdown, the focus indicator was only obscured at the bottom, where the bottom border changes on focus. The solution you describe above sounds reasonable to me. I'd say let's definitely do this for buttons. I'm not sure if it's necessary for dropdowns. If implementing a similar solution for dropdowns is clean, I'd say go for it. If for whatever reason it adds a significant amount of lines or complexity to the css for dropdowns, then I think it's worth discussing with the team. 

I'm not seeing any default browser indicator in Edge for inputs, other than a very blinky cursor during the first second of focus. I think it's safe to focus just on components that use button elements for this issue.

**Comment by mcoker** [comment]
I removed the `&::-moz-focus-inner` styles from our base.scss and everything looks OK to me. I'll submit a PR and hopefully others can have a look around and see if they notice any issues.

Re: the outline ring in edge, thinking about it more, the negative z-index on the border/background won't work as that will push the border below anything on the page that doesn't have a stacking context. So if the button is on top of an element with a white background, but that element doesn't have a stacking context, the border will appear behind the background, invisible to the user. I talked to @mceledonia and he said that a universal focus state is still in the works. Since the border-width changes when you focus on secondary and tertiary buttons, I propose we just go with that until we have a universal focus indicator. What do you think? I'll go ahead and submit a PR. Feel free to leave comments there, too.
